src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -209,19 +205,26 @@ rpmostree_sysroot_upgrader_initable_init (GInitable        *initable,
   if (!parse_origin_deployment (self, self->origin_merge_deployment, cancellable, error))
     goto out;
 
+  merge_deployment_csum =
+    ostree_deployment_get_csum (self->origin_merge_deployment);
+
   /* Now, load starting base/final checksums.  We may end up changing
    * one or both if we upgrade.  But we also want to support redeploying
    * without changing them.
    */
-  if (rpmostree_origin_is_locally_assembled (self->origin))
-    {
-      self->final_revision = g_strdup (ostree_deployment_get_csum (self->origin_merge_deployment));
-      if (!commit_get_parent_csum (self->repo, self->final_revision, &self->base_revision, error))
-        goto out;
-      g_assert (self->base_revision);
-    }
+  if (!rpmostree_deployment_get_layered_info (self->repo,
+                                              self->origin_merge_deployment,
+                                              &is_layered, &self->base_revision,
+                                              NULL, error))
+    goto out;
+
+  if (is_layered)
+    /* base layer is already populated above */
+    self->final_revision = g_strdup (merge_deployment_csum);
   else
-    self->base_revision = g_strdup (ostree_deployment_get_csum (self->origin_merge_deployment));
+    self->base_revision = g_strdup (merge_deployment_csum);
```

Comment:
`rpmostree_deployment_get_layered_info()` will only populate `base_revision` if `is_layered` is `TRUE`. So I think we're fine, right?